java/spark-1.5.2/invoke.scala

```diff
@@ -35,6 +35,9 @@ class Invoke {
               // The case that the parameter type is a Scala Seq and the argument
               // is a Java array is considered matching. The array will be converted
               // to a Seq later if this method is matched.
+            } else if (parameterTypes(i).isAssignableFrom(classOf[Array[Float]]) &&
+              (args(i) != null && args(i).getClass == classOf[Array[java.lang.Float]])) {
```

Comment:
Note to myself: while `java.lang.Float` gets implicitly (?) converted to `scala.Float` in Scala, there is no direct conversion from `Array[java.lang.Float]` to `Array[scala.Float]`.